support-services/src/test/resources/catalog-uat.json

```diff
@@ -2821,6 +2821,83 @@
       "Tier__c": null,
       "SuspensionAllowed__c": "False",
       "productRatePlans": [
+        {
+          "id": "2c92c0f971c65df50171dfabef87093d",
+          "status": "Active",
+          "name": "Corporate Digital Subscription",
+          "description": "",
+          "effectiveStartDate": "2020-01-01",
+          "effectiveEndDate": "2050-01-01",
+          "TermType__c": "TERMED",
+          "FrontendId__c": null,
+          "Saving__c": null,
+          "DefaultTerm__c": "12",
+          "RatePlanType__c": "Base",
+          "PromotionCode__c": null,
+          "TrialPeriodDays__c": null,
+          "productRatePlanCharges": [
+            {
+              "id": "2c92c0f871c64cea0171dfacb1696aad",
+              "name": "Corporate Digital Subscription",
+              "type": "OneTime",
```

Comment:
I thought subscribers had to be covered by a charge at all times in our model?  I guess you discussed this and all the systems will handle it?  I know we don't have a fulfilment process for this so it may be different - uses members data api instead.

Reply:
Yes this appears to work, subs are created and picked up by members-data-api. I went with a one time charge rather than recurring as it seemed wasteful to bill £0 on a recurring basis

Reply:
check the data lake too, I think the raw->clean subscriptions from zuora job might make that assumption.  I know I had a fun time rewriting that job when GDPR came in (retention issues)

Reply:
it looks like as long as the effective end date of the charge is populated, it will work OK.  The exception is Friend, which was a one day charge, which ends when the next one starts.
https://github.com/guardian/ophan-data-lake/blob/418ad335c5dce925a1632ca07237b79b3483f485/etl/subscriptions/src/main/scala/ophan/etl/subscriptions/JoinedToClean.scala#L135
So it may be a new exception if we want to model it like friend, or we may want to make it a charge that renews with the sub, even if it's a one time charge.

The job I linked above is due to be rewritted to use the clean zuora incremental tables, and the friend special case may be removed as I don't know if friends still exist.

Reply:
you can see the rate plan name and product name in the DL are by the following logic
https://github.com/guardian/ophan-data-lake/blob/418ad335c5dce925a1632ca07237b79b3483f485/etl/subscriptions/src/main/scala/ophan/etl/subscriptions/JoinedToClean.scala#L117-L122
First find one within the effective dates including special case for friend
failing that find the one started most recently (this was a last attempt because there were lots of GW migrated in with charges that were messed up - however this should find the corporate one time charges)
once that process has finished, find the most expensive of the ones in the list (there would only be one for corporate and it would be free therefore the most expensive)